### ISSUE TITLE
Rule configuration discoverability

### ISSUE THREAD

**Issue by idrougge** [realm/SwiftLint#2808]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version (0.33.1)
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `README.md` file specifies that rules may have custom configurability, but those options aren't listed in `Rules.md`, neither are they when executing `swiftlint rules rule_name`, rendering only the same contents as in `Rules.md`. Running `swiftlint rules` spits out a list of all rules, with a `configuration` column that hints at configurability but is truncated so that even the names of the config options cannot be read, let alone what parameters they expect.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint rules type_name

Type Name (type_name): Type name should only contain alphanumeric characters, start with an uppercase character and span between 3 and 40 characters in length.

Triggering Examples (violation is marked with '↓'):

Example #1

    class ↓myType {}

Example #2

    class ↓_MyType {}

Example #3

    private class ↓MyType_ {}

Example #4

    class ↓My {}

Example #5

    class ↓AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA {}

Example #6

    struct ↓myType {}

Example #7

    struct ↓_MyType {}

Example #8

    private struct ↓MyType_ {}

Example #9

    struct ↓My {}

Example #10

    struct ↓AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA {}

Example #11

    enum ↓myType {}

Example #12

    enum ↓_MyType {}

Example #13

    private enum ↓MyType_ {}

Example #14

    enum ↓My {}

Example #15

    enum ↓AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA {}

Example #16

    typealias ↓X = Void

Example #17

    private typealias ↓Foo_Bar = Void

Example #18

    private typealias ↓foo = Void

Example #19

    typealias ↓AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA = Void

Example #20

    protocol Foo {
      associatedtype ↓X
    }

Example #21

    protocol Foo {
      associatedtype ↓Foo_Bar: Equatable
    }

Example #22

    protocol Foo {
      associatedtype ↓AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
    }
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.33.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
Homebrew
* Paste your configuration file:
Output is the same regardless of if there is a file or not.
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
No.
* Which Xcode version are you using (check `xcode-select -p`)?
10.2.1

**Comment by msalmonse** [comment]
I have found that the rules command displays the configurations but on a reasonable sized terminal that column is truncated.

**Comment by bdrelling** [comment]
Really desperate for this on our end. SwiftLint can be hard to maintain and update at times given the lack of rule and configuration discoverability. Our current process is basically to review new SwiftLint releases and audit our .swiftlint.yml file accordingly. There’s probably so much we aren’t configuring or using properly. 

Thanks for all the hard work!

**Comment by jpsim** [comment]
Addressed in #3016.
